### PR TITLE
Feature/basic search by title

### PR DESCRIPTION
**Pull request by martynaspuskunigis**

### CODE REVIEW

File: app/db/data.ts [hunk 1]
```diff
@@ -0,0 +1,332 @@
+import { Recipe } from "./../models/Recipe";
+
+export const data: Recipe[] = [
```
Comment (DonatasBrazionis) [inline app/db/data.ts]: Use simple `.json` file. Not `.ts`. And when create methods for read/write/update/delete data from file.

File: app/controllers/index.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
-/* app/controllers/index.ts */
-export * from './welcome.controller';
\ No newline at end of file
+export * from './welcome/welcome.controller';
```
Comment (DeividasBakanas) [inline app/controllers/index.ts]: Use single kind of quotation marks.
Check you tslint.

File: app/controllers/search/search.controller.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+router.get("/:searchQuery", (req: Request, res: Response) => {
+  let { searchQuery } = req.params;
+  res.send(searchForRecipesByTitle(searchQuery));
+});
+
+function searchForRecipesByTitle(searchQuery: string): string {
+  const filteredRecipesForSearch = data.filter(recipe =>
+    recipe.title.toLowerCase().includes(searchQuery)
+  );
+  console.log(filteredRecipesForSearch);
```
Comment (DeividasBakanas) [inline app/controllers/search/search.controller.ts]: Is this `console.log` necessary?

File: app/controllers/search/search.controller.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import { getDataFromDatabase } from "../../helpers";
+
+const router: Router = Router();
+
+router.get("/", (req: Request, res: Response) => {
+  res.send(getDataFromDatabase()[0].title);
+});
+
+router.get("/rId/:id", (req: Request, res: Response) => {
```
Comment (EimD) [inline app/controllers/search/search.controller.ts]: What kind of route is `/rId/:id` ?

File: app/controllers/search/search.controller.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+function searchForRecipesByTitle(searchQuery: string): string {
+  const filteredRecipesForSearch = getDataFromDatabase().filter(recipe =>
+    recipe.title.toLowerCase().includes(searchQuery)
+  );
+  return JSON.stringify(filteredRecipesForSearch);
+}
+
+function searchForRecipesById(id: string): string {
+  const recipeFoundById = getDataFromDatabase().filter(recipe => recipe.recipe_id === id);
```
Comment (DeividasBakanas) [inline app/controllers/search/search.controller.ts]: Create separate helpers `database-helpers.ts` file for getting a data from database.

File: app/controllers/search/search.controller.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+router.get("/title/:searchQuery", (req: Request, res: Response) => {
+  const { searchQuery } = req.params;
+  res.send(searchForRecipesByTitle(searchQuery));
+});
+
+router.get("/title", (req: Request, res: Response) => {
+  res.send(getDataFromDatabase().splice(0, 10));
+});
+
+function searchForRecipesByTitle(searchQuery: string): string {
```
Comment (EimD) [inline app/controllers/search/search.controller.ts]: Move to separate file.

File: app/controllers/search/search.controller.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  const filteredRecipesForSearch = getDataFromDatabase().filter(recipe =>
```
Comment (EimD) [inline app/controllers/search/search.controller.ts]: Move to separate file.

File: app/controllers/search/search.controller.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    recipe.title.toLowerCase().includes(searchQuery)
+  );
+  return JSON.stringify(filteredRecipesForSearch);
+}
+
+function searchForRecipesById(id: string): string {
```
Comment (EimD) [inline app/controllers/search/search.controller.ts]: Move to separate file.

File: app/server.ts [hunk 9]
```diff
@@ -1,21 +1,17 @@
...
+app.use("/search", SearchController);
 
-// Serve the application at the given port
 app.listen(port, () => {
-    // Success callback
-    console.log(`Listening at http://localhost:${port}/`);
-});
\ No newline at end of file
+  // tslint:disable-next-line:no-console
+  console.log(`Listening at http://localhost:${port}/`);
```
Comment (EimD) [inline app/server.ts]: Try `console.info` instead of `console.log`.

**Pull request merged**